diff --git a/src/videosource.cpp b/src/videosource.cpp
index 10d0045..f5b5dcb 100644
--- a/src/videosource.cpp
+++ b/src/videosource.cpp
@@ -2434,10 +2434,15 @@ void BestVideoSource::WriteTimecodes(const std::filesystem::path &TimecodeFile)
         throw BestSourceException("Failed to write timecode file");
     for (const auto &Iter : TrackIndex.Frames) {
         double timestamp = ((Iter.PTS * VP.TimeBase.Num) / (double)VP.TimeBase.Den) * 1000;
+#ifdef __cpp_lib_to_chars
         char buffer[100];
         auto res = std::to_chars(buffer, buffer + sizeof(buffer), timestamp, std::chars_format::fixed, 2);
         if (fprintf(F.get(), "%s\n", std::string(buffer, res.ptr - buffer).c_str()) < 0)
             throw BestSourceException("Failed to write timecode file");
+#else
+        if (fprintf(F.get(), "%.02f\n", timestamp) < 0)
+            throw BestSourceException("Failed to write timecode file");
+#endif
     }
     if (!CloseWrittenFile(F))
         throw BestSourceException("Failed to write timecode file");
